### PR TITLE
[CPU Plugin 2.0] Remove IE extension related

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - Remove IE extension related
 - Remove extension manager

### Tickets:
 - CVS-124099

### CODE REVIEW

File: src/plugins/intel_cpu/src/node.cpp [hunk 1]
```diff
@@ -1302,16 +1301,11 @@ Node* Node::NodesFactory::create(const std::shared_ptr<ngraph::Node>& op, const
...
-        std::unique_ptr<Node> ol(createNodeIfRegistered(intel_cpu, Type::Generic, op, context));
-        if (ol != nullptr && ol->created(context->getExtensionManager()))
-            newNode = ol.release();
-    }
 
     if (newNode == nullptr) {
         try {
             std::unique_ptr<Node> ol(createNodeIfRegistered(intel_cpu, TypeFromName(op->get_type_name()), op, context));
-            if (ol != nullptr && ol->created(context->getExtensionManager()))
+            if (ol != nullptr)
```
Comment (riverlijunjie) [inline src/plugins/intel_cpu/src/node.cpp:1308]:
```suggestion
            if (ol != nullptr && ol->created())
```
We still need to check "ol->created()".

File: src/plugins/intel_cpu/src/node.cpp [hunk 2]
```diff
@@ -1325,7 +1319,7 @@ Node* Node::NodesFactory::create(const std::shared_ptr<ngraph::Node>& op, const
     if (newNode == nullptr) {
         try {
             std::unique_ptr<Node> ol(new Reference(op, context, errorMessage));
-            if (ol != nullptr && ol->created(context->getExtensionManager()))
+            if (ol != nullptr)
```
Comment (riverlijunjie) [inline src/plugins/intel_cpu/src/node.cpp:1322]:
```suggestion
           if (ol != nullptr && ol->created())
```

File: src/plugins/intel_cpu/src/extension.cpp [hunk 3]
```diff
@@ -2,197 +2,117 @@
...
-}
-
-}   // namespace intel_cpu
-}   // namespace ov
-
-// Generate exported function
-IE_DEFINE_EXTENSION_CREATE_FUNCTION(ov::intel_cpu::Extension)
+        // type relaxed opset
+        std::make_shared<ov::OpExtension<ov::op::TypeRelaxed<ngraph::op::v1::Add>>>(),
+        std::make_shared<ov::OpExtension<ov::op::TypeRelaxed<ngraph::op::v1::AvgPool>>>(),
```
Comment (riverlijunjie) [inline src/plugins/intel_cpu/src/extension.cpp:41]: We'd better not use `ngraph::op::`, but use `ov::op::`

**Comment by ilya-lavrenov** [comment]
Maybe we can also remove this method https://github.com/openvinotoolkit/openvino/blob/master/src/inference/include/openvino/runtime/core.hpp#L362-L370 and similar from API 1.0, otherwise users can provide old API extension, but it's ignored by CPU plugin who is the only user of old extensions

**Comment by ilya-lavrenov** [comment]
Let's resolve conflicts and merge

**Comment by zhaixuejun1993** [comment]
Rewrite this PR, caused by too many conflict & big changes!
Replaced by https://github.com/openvinotoolkit/openvino/pull/21915

**Pull request closed without merging**